torchaudio/datasets/utils.py

```diff
@@ -234,6 +236,7 @@ def __len__(self) -> int:
         return len(self.dataset)
 
 
+@deprecated('Please migrate to torchdata.', version='0.11')
```

Comment:
actually, could we replace this with a better message? [torchdata](https://github.com/pytorch/data) is still a prototype depending on pytorch nightly currently

Reply:
In fact, there is no need to replace this because these utilities do not provide any value to end user.
Specifically, `bg_iterator` is reported to cause deadlock https://github.com/pytorch/audio/issues/733#issuecomment-736092030
So they should not be used at all. It is not documented either, so I bet no one is using it.
Since this is not a public API, and not documented I actually think the right way is to simply delete them. (won't be BC-breaking)
This was a sort of compromise. If the current message is not helpful, it will be strange to say `'Please stop using'` either. We can just delete them too. What do you think?

Reply:
Is there any downside in not including an explicit message and just simply saying that it will be deprecated in v0.11? That way, we still provide warning to anyone who may be using the functions (even if unlikely), but do not provide a misleading/unnecessary message.